pkg/adminconfig/rego_file_evaluator.go

```diff
@@ -58,27 +85,43 @@ func (r *RegoPolicyEvaluator) prepareQuery() (rego.PreparedEvalQuery, error) {
 			continue
 		}
 		fileName := filepath.Join(RegoPolicyDirectory, name)
-		module, err := ioutil.ReadFile(filepath.Clean(fileName))
+		module, err := os.ReadFile(filepath.Clean(fileName))
 		if err != nil {
 			return rego.PreparedEvalQuery{}, err
 		}
 		modules[name] = string(module)
 	}
+	r.Timestamp = time.Now()
 	compiler, err := ast.CompileModules(modules)
```

Comment:
We recompile modules when we are doing a query and discover that the policy files have been updated according to the code here.  (1) I assume this could have a significant performance impact on this particular query.  Is this the best approach to take?  Perhaps this should be done outside of the query path?  (2) This same issue is being discussed about data governance policies.  There we are looking at creating an event when policies change, and having fybrikapplication reconcile run again to see if any changes are required to the data plane - assuming that changes should affect existing workloads and their data planes.  (3) Since changes to existing data planes may not always be required, perhaps we need some indication in the changed policies as to whether they should affect existing workloads.

Reply:
(2) and (3) are about existing data planes. If a data plane exists and the spec of the application did not change, we won't do nothing. Whether this is the right approach or not is a bigger discussion but unrelated to this PR.
(1) Performance impact will be only if policies have changed, which does not happen often.